src/web/src/pages/Main.vue

```diff
@@ -118,20 +118,11 @@ export default {
       ICS_DAY_SHORTNAMES: ['MO', 'TU', 'WE', 'TH', 'FR', 'SA', 'SU']
     };
   },
-  created() {
-    if (this.$route.query.semester) {
-      this.currentSemester = this.$route.query.semester;
-    } else {
-      getDefaultSemester().then(semester => {
-        this.currentSemester = semester;
-      });
-    }
+  async created() {
```

Comment:
since we're very much operating as a SPA and we're passing currentSemester via props, can probably change footer to emit changes to `currentSemester` and avoid reloading the whole page

Reply:
that makes sense, I'll look into it.

Reply:
Ok, I've done it the way you described. Because course fetching is parameterized by currentSemester each time you change semester a new API call occurs. Wonder if ppl will change semester enough for this to be a problem?? Hopefully not.